### PR TITLE
debug: remove type-specific endpoints

### PR DESCRIPTION
**Pull request by howardjohn**
This has been a part of config_dump for many releases, we should
consolidate one this

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -187,9 +186,6 @@ func (s *DiscoveryServer) AddDebugHandlers(mux, internalMux *http.ServeMux, enab
 		s.addDebugHandler(mux, internalMux, "/debug/force_disconnect", "Disconnects a proxy from this Pilot", s.forceDisconnect)
 	}
 
-	s.addDebugHandler(mux, internalMux, "/debug/ecdsz", "Status and debug interface for ECDS", s.ecdsz)
-	s.addDebugHandler(mux, internalMux, "/debug/edsz", "Status and debug interface for EDS", s.Edsz)
-	s.addDebugHandler(mux, internalMux, "/debug/ndsz", "Status and debug interface for NDS", s.ndsz)
```
Comment (keithmattix) [inline pilot/pkg/xds/debug.go:192]: I think I can get behind ecds and eds, but isn't NDS completely within pilot-agent and not envoy? Just confirming that it is in fact returned in `config_dump` i.e. `istioctl pc all`

Comment (keithmattix) [reply pilot/pkg/xds/debug.go:192]: Good point this is just pilot; I'm good with that then

**Comment by howardjohn** [comment]
> Does it cost much? i see many developers depend on nds eds, not much use ecds

the original motivation was fixing the race conditions stuff.. I noticed this uses a misnamed function ('Clusters" when it's really endpoints) so I went from there. It's entirely redundant with config dump so maybe at best we refactor it to use the same logic..?

File: pilot/pkg/xds/debug.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:192]: I think having /ndsz in pilot is useful.

Please add release notes as there may be folks using these endpoints in some internal tools

**Comment by hzxuzhonghu** [comment]
Yea, normally it is same as configdump. I mean this used when they find the proxy contains unexpected xds configuration, users want to make sure if it is because envoy unsync with pilot

**Review by howardjohn — COMMENTED** [review]
OK changed the PR to keep the APIs but to implement them in terms of `config_dump`

**Review by keithmattix — APPROVED** [review]
Assuming we tests these endpoints and the tests pass, this is good with me

**Pull request merged**